filament/backend/src/webgpu/platform/WebGPUPlatform.cpp

```diff
@@ -39,6 +54,185 @@ namespace filament::backend {
 
 namespace {
 
+constexpr std::array REQUIRED_FEATURES = {
+    wgpu::FeatureName::TransientAttachments };
+
+constexpr std::array OPTIONAL_FEATURES = {
+    wgpu::FeatureName::DepthClipControl,
+    wgpu::FeatureName::Depth32FloatStencil8,
+    wgpu::FeatureName::CoreFeaturesAndLimits };
+
+enum class LimitToValidate : uint8_t {
+    MAX_BIND_GROUPS = 0,
+    MAX_BINDINGS_PER_BIND_GROUP = 1,
+    MAX_SAMPLERS_PER_SHADER_STAGE = 2,
+    MAX_STORAGE_BUFFERS_PER_SHADER_STAGE = 3,
+    MAX_VERTEX_BUFFERS = 4,
+    MAX_VERTEX_ATTRIBUTES = 5,
+    UNKNOWN = 6// needs to be last for iterating through all possible limits in the enum
+};
+
+constexpr wgpu::Limits REQUIRED_LIMITS = wgpu::Limits{
+    .maxBindGroups = MAX_DESCRIPTOR_SET_COUNT,
```

Comment:
why are you using this value for `maxBindGroups` and `maxBindingsPerBindGroup`? This is confusing given that you just defined `MAX_BIND_GROUPS` in the `LimitToValidate enum` above.